corehq/form_processor/interfaces/processor.py

```diff
@@ -142,31 +143,35 @@ def xformerror_from_xform_instance(self, instance, error_message, with_new_id=Fa
         return self.processor.xformerror_from_xform_instance(instance, error_message, with_new_id=with_new_id)
 
     def update_responses(self, xform, value_responses_map, user_id):
+        '''
+        Update a set of question responses. Returns a list of any
+        questions that were not found in the xform.
+        '''
         from corehq.form_processor.utils.xform import update_response
+        errors = []
 
         xml = xform.get_xml_element()
-        dirty = False
         for question, response in six.iteritems(value_responses_map):
-            if update_response(xml, question, response, xmlns=xform.xmlns):
-                dirty = True
-
-        if dirty:
-            from couchforms.const import ATTACHMENT_NAME
-            from corehq.form_processor.interfaces.dbaccessors import FormAccessors
-            from corehq.form_processor.models import Attachment
-
-            existing_form = FormAccessors(xform.domain).get_with_attachments(xform.get_id)
-            existing_form, new_form = self.processor.new_form_from_old(existing_form, xml,
-                                                                       value_responses_map, user_id)
-            new_xml = etree.tostring(xml)
-            interface = FormProcessorInterface(xform.domain)
-            interface.store_attachments(new_form, [
-                Attachment(name=ATTACHMENT_NAME, raw_content=new_xml, content_type='text/xml')
-            ])
-            interface.save_processed_models([new_form, existing_form])
-            return True
-
-        return False
+            try:
+                update_response(xml, question, response, xmlns=xform.xmlns)
+            except XFormQuestionValueNotFound:
+                errors.append(question)
+
+        from couchforms.const import ATTACHMENT_NAME
+        from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+        from corehq.form_processor.models import Attachment
```

Comment:
the const import could be moved to the top of the file to clean up this method. Not sure if the others would cause a cyclic import